Detectors/TRD/raw/src/TrackletsParser.cxx

```diff
@@ -0,0 +1,149 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   TrackletParser.h
+/// @brief  TRD raw data parser for Tracklet dat format
+
+#include "TRDRaw/TrackletsParser.h"
+#include "DataFormatsTRD/RawData.h"
+#include "DataFormatsTRD/Tracklet64.h"
+
+#include "fairlogger/Logger.h"
+
+//TODO come back and figure which of below headers I actually need.
+#include <cstring>
+#include <string>
+#include <vector>
+#include <array>
+
+namespace o2
+{
+namespace trd
+{
+
+int TrackletsParser::Parse()
```

Comment:
This reminds me quite a bit of the issues we had when I looked into this with Toko... Not an easy one to do elegantly. Let me know if we should chat about it.

The same comment applies to the digits reader, too, I assume.

Reply:
I am not sure I follow, but sure, mattermost,zoom,vidyo I am available till about midnight maybe a bit longer.